packages/builder/src/schemas.zod.ts

```diff
@@ -0,0 +1,304 @@
+import { z } from 'zod';
+import { handleZodErrors } from './error/zod';
+
+/// ================================ INPUT CONFIG SCHEMAS ================================ \\\
+
+export const contractSchema = z
```

Comment:
so for the builder base cannon steps its all well to put the zod definitions in this file. However, what happens for custom steps for validation (ex. `cannon-plugin-router` https://github.com/Synthetixio/cannon-plugin-router )

I *personally* am not a big fan of putting all the types for a package into a single file like some sort of a dumping ground

Reply:
Hmmm thats a good point. Existing plugins will have to switch their schemas to zod. However its worth rethinking a few things based on this:

- Before migrating, we validated any incoming "action" config schema as a JSON schema with properties and optional properties, any parameter was valid as long as it followed basic JTDDataType guidelines.          
- I realized that the ConfigValidationSchema that is used to validate any action schema the way I wrote it is probably not gonna work, since we need to sort of be able to do a "catchall" on config properties created for custom plugins.

My suggestion for that bug is to change the ConfigValidationSchema to be way less strict and see if there's a way for us to allow for new parameters in custom actions.

As for putting everything in one file: I don't personally like it either, it just makes generating the docs a bit easier but i can achieve the same output with a little extra work if we prefer to have the types in their respective action file.